src/paginators/base-paginator-spec.js

```diff
@@ -0,0 +1,361 @@
+
+        return expect(requestCallCount).to.eventually.equal(1);
+      });
+    });
+
+    describe('prev', function() {
+      beforeEach(function() {
+        makePaginator({ page: 2 });
+      });
+
+      it('returns a promise that resolves', function() {
+        const promise = paginator.prev();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with a response object', function() {
+        const response = paginator.prev();
+
+        return expect(response).to.eventually.be.an('object');
+      });
+
+      it('decrements the page', function() {
+        expect(paginator.page).to.equal(2);
+
+        const page = paginator.prev()
+          .then(() => paginator.page);
+
+        return expect(page).to.eventually.equal(1);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.prev()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(1);
+      });
+    });
+
+    describe('first', function() {
+      beforeEach(function() {
+        makePaginator();
+      });
+
+      it('returns a promise that resolves', function() {
+        const promise = paginator.first();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with a response object', function() {
+        const response = paginator.first();
+
+        return expect(response).to.eventually.be.an('object');
+      });
+
+      it('sets the page to the first page ', function() {
+        expect(paginator.page).to.equal(0);
+
+        const page = paginator.first()
+          .then(() => paginator.page);
+
+        return expect(page).to.eventually.equal(1);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.first()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(1);
+      });
+    });
+
+    describe('last', function() {
+      beforeEach(function() {
+        makePaginator();
+      });
+
+      it('returns a promise that resolves', function() {
+        const promise = paginator.last();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with a response object', function() {
+        const response = paginator.last();
+
+        return expect(response).to.eventually.be.an('object');
+      });
+
+      it('sets the page to the last page', function() {
+        expect(paginator.page).to.equal(0);
+
+        const page = paginator.last()
+          .then(() => paginator.page);
+
+        return expect(page).to.eventually.equal(10);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.last()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(2);
```

Comment:
Why is this 2 as opposed to 1? I'm guessing it's because paginator requests the first page (or whatever is set in the options) before making a request for the last page. Is this correct?